File: tests/test_changepoints.R
library(changepointsHD)

set.seed(334)

scp_data = read.table("scp.txt")
scp_data = as.matrix(scp_data)

# prox gradient black-box method
cov_est = cov(scp_data)
init = solve(cov_est)
res_map = prox_gradient_mapping(scp_data, init, 0.1, 0.99, 0.1, 100, 1e-20)

# prox gradient black-box ll
res_ll = prox_gradient_ll(scp_data, res_map, 0.1)

prox_gradient_params=list()
prox_gradient_params$update_w = 0.1
prox_gradient_params$update_change = 0.99
prox_gradient_params$regularizer = 0.1
prox_gradient_params$max_iter = 1
prox_gradient_params$tol = 1e-5

prox_gradient_ll_params=list()
prox_gradient_ll_params$regularizer = 0.1

simulated_annealing_params = list()
simulated_annealing_params$buff=10

# test simulated annealing
changepoint_mod = changepointMod(bbmod=prox_gradient_mapping,
                                 log_likelihood=prox_gradient_ll,
                                 bbmod_params=prox_gradient_params,
                                 ll_params=prox_gradient_ll_params,
                                 part_values=list(init, init),
                                 data=list(scp_data))
changepoint_mod = simulated_annealing(changepoint_mod, buff=10)
stopifnot(changepoint_mod@changepoints == 56)

# test brute force
changepoint_mod@part_values = list(init, init)
changepoint_mod = brute_force(changepoint_mod, buff=10)
stopifnot(changepoint_mod@changepoints == 77)

# test rank one
res_ro = rank_one(scp_data, init, update_w=0.1, regularizer=0.1)
stopifnot(res_ro$tau == 56)

mcp_data = read.table("mcp.txt")
mcp_data = as.matrix(mcp_data)

# test binary segmentation
changepoint_mod@part_values = list(init, init)
changepoint_mod = binary_segmentation(changepoint_mod, method=simulated_annealing,
                                      thresh=0, buff=10,
                                      method_params=simulated_annealing_params)
stopifnot(changepoint_mod@changepoints == c(0, 18, 35, 56, 68, 85, 100))

File: tests/test_changepoints.py
from datetime import datetime
from changepoints import *
import numpy as np

t1 = datetime.now()

np.random.seed(334)

scp_data = np.loadtxt("scp.txt")

# prox gradient black-box method
cov_est = np.cov(scp_data.T)
init = np.linalg.inv(cov_est)
res_map = mapping(scp_data, init, 0.1, 0.99, 0.1, 100, 1e-4)

# prox gradient black-box ll
res_ll = log_likelihood(scp_data, res_map, 0.1)

# simulated annealing with prox gradient
method_kwds = {"update_w": 0.1, "update_change": 0.99,
        "regularizer": 0.1, "mapping_iter": 1,
        "tol": 1e-5}
ll_kwds = {"regularizer": 0.1}
res_sa = simulated_annealing(scp_data, init, mapping, log_likelihood, buff=10,
                             method_kwds=method_kwds, ll_kwds=ll_kwds)

# brute force with prox gradient
res_bf = brute_force(scp_data, init, mapping, log_likelihood, buff=10,
                     method_kwds=method_kwds, ll_kwds=ll_kwds)

# rank one with prox gradient
res_ro = rank_one(scp_data, init, regularizer=0.1, update_w=0.1, buff=10,
                  max_iter=5)


t2 = datetime.now()
print t2 - t1
#res = simulated_annealing(scp_data, Tmin=1./100, Tmax=1., regularizer=0.1,
#                          update_w=0.1, buff=10, max_iter=500)

#res = rank_one(scp_data, regularizer=0.1, update_w=0.1, buff=10, max_iter=5)

#res = brute_force(scp_data, buff=10, mapping_iter=5)

#mcp_data = np.loadtxt("mcp.txt")

#method_kwds = {"regularizer": 0.1, "update_w": 0.1}
#res = binary_segmentation(mcp_data, 0, simulated_annealing, 10,
#                          method_kwds=method_kwds)
